packages/legend-query-builder/src/stores/QueryBuilderTypeaheadHelper.ts

```diff
@@ -54,6 +69,57 @@ const initializeQueryBuilderState = (
   return builderState;
 };
 
+const applyMilestoningDates = (
+  propertyExpression: AbstractPropertyExpression,
+  observerContext: ObserverContext,
+): void => {
+  let currentExpression: ValueSpecification | undefined = propertyExpression;
+  while (currentExpression instanceof AbstractPropertyExpression) {
+    // check if the property is milestoned
+    if (
+      currentExpression.func.value.genericType.value.rawType instanceof Class &&
+      currentExpression.func.value._OWNER._generatedMilestonedProperties
+        .length !== 0 &&
+      currentExpression.func.value instanceof DerivedProperty &&
+      !currentExpression.func.value._OWNER.derivedProperties.includes(
+        currentExpression.func.value,
+      )
+    ) {
+      const parameterValue = createSupportedFunctionExpression(
+        QUERY_BUILDER_SUPPORTED_FUNCTIONS.NOW,
+        PrimitiveType.DATETIME,
+      );
+      const parameterValues =
+        currentExpression.parametersValues.length === 2
+          ? [parameterValue]
+          : [parameterValue, parameterValue];
+      functionExpression_setParametersValues(
```

Comment:
i have a major concern here
arent we replacing the parameter value of this derived property in real time. so we are changing the actual value of the user's query when we create the typeahead query. is that correct?

Reply:
I missed this. Thanks for pointing this!! This would actually modify the existing values. I think will create a new AbstractPropertyExpression instead of modifying it.

Reply:
Updated the test cases to catch such scenarios